### PR TITLE
cleanup(kubelet): replace context.TODO to context.Background in Kubelet.Run and NewMainKubelet

### PR DESCRIPTION
**Pull request by soma00333**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
We are implementing [Structured and Contextual Logging](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md).
In [PR#127887](https://github.com/kubernetes/kubernetes/pull/127887), @bart0sh mentioned about further work on pkg/kubelet/kubelet.go: [cf](https://github.com/kubernetes/kubernetes/pull/127887#discussion_r1797749572)
This PR replaces context.TODO() in Kubelet.Run and NewMainKubelet.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
Support for contextual logging in other parts of pkg/kubelet will be addressed in other PRs. 
The scope of this PR is only Kubelet.Run and NewMainKubelet.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -241,7 +241,7 @@ type Bootstrap interface {
 	ListenAndServe(kubeCfg *kubeletconfiginternal.KubeletConfiguration, tlsOptions *server.TLSOptions, auth server.AuthInterface, tp trace.TracerProvider)
 	ListenAndServeReadOnly(address net.IP, port uint, tp trace.TracerProvider)
 	ListenAndServePodResources()
-	Run(<-chan kubetypes.PodUpdate)
+	Run(ctx context.Context, updates <-chan kubetypes.PodUpdate)
```
Comment (rphillips) [inline pkg/kubelet/kubelet.go:244]: There is another PR about adding contexts as well. This function should probably return an error that captures if the context is ever cancelled or times out.

**Review by rphillips — CHANGES_REQUESTED** [review]

File: pkg/kubelet/kubelet.go (hunk 1 above)
Comment (soma00333) [reply pkg/kubelet/kubelet.go:244]: I appreciate your comment👍
Should I close this one if there is already a PR for adding contexts?

Comment (rphillips) [reply pkg/kubelet/kubelet.go:244]: Perhaps. This PR [1] is adding context other places... I don't think there is an overlap, but the comments are probably relevant.

[1] https://github.com/kubernetes/kubernetes/pull/128008

Comment (soma00333) [reply pkg/kubelet/kubelet.go:244]: Thank you👍
I closed this PR and dropped a comment at https://github.com/kubernetes/kubernetes/pull/127887#discussion_r1798635680

**Pull request closed without merging**